File: designPattern/visitor/test1/StringVisitor.java
import java.util.Iterator;

public class StringVisitor extends Visitor{
	String currentFolderName = "";
	String savedName;
	public void visit(File file){
		System.out.println(currentFolderName+"/"+file);
	}
	public void visit(Directory directory){
		System.out.println(currentFolderName+"/"+directory);
		Iterator iterator = directory.iterator();
		savedName = currentFolderName;
		currentFolderName = currentFolderName+"/"+directory.getName();
		while(iterator.hasNext()){
			Entry entry = (Entry)iterator.next();
			entry.accept(this);
		}
		currentFolderName = savedName;
	}
}

File: designPattern/visitor/pythonbasic/StringVisitor.py
from Visitor import Visitor

class StringVisitor(Visitor):

    def __init__(self):
        self.prefix = ""
        self.save = ""

    def visitFile(self, file):
        print(self.prefix+"/"+file.getName())

    def visitDirectory(self, directory):
        print(self.prefix+"/"+directory.getName())
        self.save = self.prefix
        self.prefix = self.prefix+"/"+directory.getName()
        for file_or_dir in directory.documents:
            file_or_dir.accept(self)
        self.prefix = self.save

